accounts/abi/bind/backends/simulated.go

```diff
@@ -750,6 +751,35 @@ func (b *SimulatedBackend) SendTransaction(ctx context.Context, tx *types.Transa
 	return nil
 }
 
+// a replica of above `SendTransaction` function
+func (b *SimulatedBackend) SendTransactionConditional(ctx context.Context, tx *types.Transaction, knownAccounts map[string]map[common.Address]interface{}) error {
+	// check knownAccounts
+	currentState, _ := b.blockchain.State()
+
+	for k, v := range knownAccounts["knownAccounts"] {
+		// check if the value is hex string or an object
+		if object, ok := v.(string); ok {
+			trie, _ := currentState.StorageTrie(k)
+			actualRootHash := trie.Hash()
```

Comment:
Bor also has a way to fetch the storage root as we did in erigon. Let's see if we can use that here.

Reply:
Removed this all together.